include/libtorrent/session.hpp

```diff
@@ -131,6 +121,44 @@ namespace libtorrent
 		boost::shared_ptr<aux::session_impl> m_impl;
 	};
 
+	struct TORRENT_EXPORT session_params
+	{
+		session_params(settings_pack sp = settings_pack()
+			, int flags = default_flags)
+			: settings(sp)
+			, start_flags(flags)
+#ifndef TORRENT_DISABLE_DHT
+			, dht_storage_constructor(dht::dht_default_storage_constructor)
+#endif
+		{}
+		session_params(session_params const&) = default;
+		session_params(session_params&&) = default;
+		session_params& operator=(session_params const&) = default;
+		session_params& operator=(session_params&&) = default;
+
+		// flags to be passed in to the constructor
+		enum session_flags_t
```

Comment:
I rather remove this enum :D ... I imagine swig struggling big time with the idea